src/components/PostPageDetail/elements/Content.js

```diff
@@ -84,12 +85,26 @@ const Content = ({
     }
   }, [containerHeight, textHeight]);
 
+  const randomIndex = React.useMemo(() => {
+    return Math.floor(Math.random() * listFeedColor.length);
+  }, []);
+
   const isShortText = () => {
     return images_url.length <= 0 && item.post_type === POST_TYPE_STANDARD && !haveSeeMore;
   };
   const handleContainerPdp = () => {
     if (isShortText()) {
-      return styles.shortText(parentData?.bg);
+      const backgroundData = (() => {
+        if (parentData?.bg) {
+          return parentData.bg;
+        }
+        if (item.anon_user_info_color_code) {
+          return hexToRgb(item.anon_user_info_color_code, 0.25);
+        }
+        const newColor = listFeedColor[randomIndex];
+        return hexToRgb(newColor.bg, 0.25);
+      })();
+      return styles.shortText(backgroundData);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic in `handleContainerPdp` function seems fine, but it might be more readable and maintainable if you separate the background color calculation into its own function. This way, `handleContainerPdp` only focuses on handling the container and the new function can handle the background color calculation.

```diff
- const handleContainerPdp = () => {
-   if (isShortText()) {
-     const backgroundData = (() => {
-       if (parentData?.bg) {
-         return parentData.bg;
-       }
-       if (item.anon_user_info_color_code) {
-         return hexToRgb(item.anon_user_info_color_code, 0.25);
-       }
-       const newColor = listFeedColor[randomIndex];
-       return hexToRgb(newColor.bg, 0.25);
-     })();
-     return styles.shortText(backgroundData);
-   }
-   return {};
- };

+ const calculateBackgroundColor = () => {
+   if (parentData?.bg) {
+     return parentData.bg;
+   }
+   if (item.anon_user_info_color_code) {
+     return hexToRgb(item.anon_user_info_color_code, 0.25);
+   }
+   const newColor = listFeedColor[randomIndex];
+   return hexToRgb(newColor.bg, 0.25);
+ };
+
+ const handleContainerPdp = () => {
+   if (isShortText()) {
+     const backgroundData = calculateBackgroundColor();
+     return styles.shortText(backgroundData);
+   }
+   return {};
+ };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->